grype-server/pkg/scanner/scanner.go

```diff
@@ -0,0 +1,188 @@
+package scanner
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"github.com/anchore/grype/grype"
+	grype_pkg "github.com/anchore/grype/grype/pkg"
+	"github.com/anchore/grype/grype/presenter/models"
+	"github.com/anchore/grype/grype/vulnerability"
+	"github.com/anchore/syft/syft"
+	"github.com/anchore/syft/syft/format"
+	log "github.com/sirupsen/logrus"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Portshift/grype-server/grype-server/pkg/rest"
+	"github.com/anchore/grype/grype/db"
+)
+
+type Config struct {
+	RestServerPort int
+	DbRootDir      string
+	DbUpdateURL    string
+}
+
+type Scanner struct {
+	restServer  *rest.Server
+	dbCurator   *db.Curator
+	DbRootDir   string
+	DbUpdateURL string
+
+	sync.RWMutex
+	vulProvider         *vulnerability.StoreAdapter
+	vulMetadataProvider *vulnerability.MetadataStoreAdapter
+}
+
+func Create(conf *Config) (*Scanner, error) {
+	var err error
+	s := &Scanner{
+		DbRootDir:   conf.DbRootDir,
+		DbUpdateURL: conf.DbUpdateURL,
+	}
+
+	s.restServer, err = rest.CreateRESTServer(conf.RestServerPort, s)
+	if err != nil {
+		return nil, fmt.Errorf("failed to start rest server: %v", err)
+	}
+
+	return s, nil
+}
+
+func (s *Scanner) Start(ctx context.Context, errChan chan struct{}) error {
+	dbConfig := &db.Config{
+		DBRootDir:          s.DbRootDir,
+		ListingURL:         s.DbUpdateURL,
+		ValidateByHashOnGet: false,
+	}
+	if err := s.loadBb(dbConfig); err != nil {
+		return fmt.Errorf("failed to load DB: %v", err)
+	}
+	s.startUpdateChecker(ctx)
+	s.restServer.Start(errChan)
+
+	return nil
+}
+
+func (s *Scanner) Stop() {
+	if s.restServer != nil {
+		s.restServer.Stop()
+	}
+}
+
+func (s *Scanner) Scan(sbom64 string) (string, error) {
+	sbom, err := base64.StdEncoding.DecodeString(sbom64)
+	if err != nil {
+		return "", fmt.Errorf("failed to decode sbom: %v", err)
+	}
```

Comment:
when configuring the format to `byte` this will not be needed, and also for the return value.